block/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -72,16 +83,67 @@ retrievalWorker keepAliveTimer = worker outs (retrievalWorkerImpl keepAliveTimer
 --
 -- If both happen at the same time, 'BlockRetrievalQueue' takes precedence.
 --
+-- Note: if we haven't caught up with the blockchain yet, the worker will
+-- try to download and apply blockchain dumps, and only if it fails it'll
+-- switch to requesting headers/blocks.
+--
 retrievalWorkerImpl
     :: forall ctx m.
        (BlockWorkMode ctx m)
-    => Timer -> SendActions m -> m ()
-retrievalWorkerImpl keepAliveTimer SendActions {..} =
-    handleAll mainLoopE $ do
-        logInfo "Starting retrievalWorker loop"
-        mainLoop
+    => Timer -> Maybe Text -> SendActions m -> m ()
+retrievalWorkerImpl keepAliveTimer blockStorageMirror SendActions {..} = do
+    handleAll tryDownloadE $ do
+        epoch <- view epochIndexL <$> DB.getTipHeader
+        whenJust blockStorageMirror $ \dumpUrl ->
+            tryDownload epoch (toString dumpUrl)
+    handleAll recoveryLoopE $ do
+        logInfo "Starting recovery loop"
+        recoveryLoop
   where
-    mainLoop = do
+    -- | Try to download and apply a blockchain dump, starting from the
+    -- given epoch.
+    tryDownload epoch dumpUrl = whenNothingM_ getCurrentSlot $ do
+        let epochUrl = dumpUrl <//> format "epoch{}.cbor.lzma"
+                                           (toInteger epoch)
+        logInfo ("Downloading blockchain dump for epoch "+|epoch|+
+                 " from "+|epochUrl|+"")
+        isSuccess <- modifyStateLock HighPriority "tryDownload" $ \tip -> do
```

Comment:
This should be high priority, otherwise it is possible for the action to be postponed until something more important happens. But recovery is a top priority.